lib/diagnostic.js

```diff
@@ -39,20 +58,33 @@ let owner = null;
 // Methods: vocalize
 
 // Replace null with your answer
-let pet = null;
+let pet = {
+  name: 'Octavia',
+  species: 'Gecko',
+  breed: 'Leopard',
+  vocalization: 'Moooooo'
+};
 
 // 4. Create a constructor function for pet objects, as described above. The
 // constructor function should be responsible for setting the values of `name`,
 // `species`, `breed`, and `vocalization`.
 
-const Pet = function () {
-  /* Your answer here */
+const Pet = function ()
+ {
+  this.name = name
+  this.species = species
+  this.breed = breed
+  this._vocalization = speak
 };
 
 // 5. Attach a `vocalize` method to the Pet's prototype.  The `vocalize` method
 // should return the pet's vocalization attribute.
 
-/* your answer here */
+const speak = function () {
```

Comment:
I think you reversed these a little bit. What you wanted to do was add a `vocalization` property above, something like `this.vocalization = vocalization`. Below, you would add a method that returns this `vocalization` like so: `vocalize: function() {return this.vocalization}`

If I wanted to use this constructor, it would look something like this: `let petExample = new Pet('Spot', 'Dog', 'Terrier', 'Woof')`. Then if I wanted to call `petExample.vocalize()`, it would return `Woof`